### ISSUE TITLE
Show most (recently) used POI types by default in Things overlay

### ISSUE THREAD

**Issue by FloEdelmann** [streetcomplete/StreetComplete#5622]
Note: This is partly related to #5621, which suggests other improvements to the [POI type selection dialog](https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/view/dialogs/SearchFeaturesDialog.kt).

**Use case**
I often map street cabinets, which are not in the [top 9 POI types](https://github.com/streetcomplete/StreetComplete/blob/113c31271411947c101f728a390beefca928eeaf/app/src/main/java/de/westnordost/streetcomplete/osm/Things.kt#L256-L282) displayed by default when adding a new POI in the Things overlay. Thus, I always have to search for the feature explicitly, which is cumbersome.

I know that not everyone is interested in street cabinets though (but some might be into guideposts, memorials, etc.), so there is no selection that fits everyone.

**Proposed Solution**
The POI types that I have most recently selected should be displayed above the top 9 POI types shown by default. We already have the [`LastPickedValuesStore`'s `mostCommonWithin` util](https://github.com/streetcomplete/StreetComplete/blob/113c31271411947c101f728a390beefca928eeaf/app/src/main/java/de/westnordost/streetcomplete/util/LastPickedValuesStore.kt#L28-L42) for this; the parameters can be tweaked. Without having it tried out yet, I'd suggest `target = 3`, `historyCount = 15`, `first = 1`.

**Comment by mnalis** [comment]
:+1: - I was actually assuming that was the case -- but I use mostly SCEE and that list definitely seems to be changing depending on the usage there (for the long-press `Add POI` variant at least).

It is often useful for me, e.g. on a path, I add mostly `trail marker`, `tree` and occasional `bench`. Having to search for each of them every time is not good UX, as there are many to be encountered in that walk.

**Comment by Helium314** [comment]
> (but I use mostly SCEE and that list definitely seems to be changing depending on the usage there)

Interestingly I don't remember adding this. Maybe it's a side effect of having this recently used list in the generic add POI dialog on long-clicking the map.

**Comment by westnordost** [comment]
This is a deliberate omission.

Having most recently selected surfaces at the top makes sense, because if one path has a rock surface, it is very much more likely that the adjacent path will also have a rock surface, etc.

Having most recently selected building levels and roof shapes easily selectable makes sense, because the building heights in one area tend to be similar.

Having the most recently selected cycleway situation easily selectable makes sense because if for a stretch of road, a particular cycleway infrastructure situation exists, it is very likely that the same exists for the adjacent sections.

---

For POIs, especially the "things", it is arguably the other way round. When the user just mapped a hydrant or a street cabinet, it somewhat reduces the chance for another POI of this type to be in the immediate vicinity.
So, it does not appear to me to be a useful mechanism to put whatever at the top of the list that the user just selected. I recognize the use case of users wanting to specifically always add certain things that are not mapped (at all) yet in his area, but the `lastPicketValuesStore` seems to be the wrong instrument for that.

**Comment by matkoniecz** [comment]
> For POIs, especially the "things", it is arguably the other way round. When the user just mapped a hydrant or a street cabinet, it somewhat reduces the chance for another POI of this type to be in the immediate vicinity.

Maybe it would be applicable if many mappers map some narrow subset of all things? So for example someone mapping only advertising features* would end with such features for easy selection.

*I know about such person though they do not use SC, they later check which are legal and report illegal for removal

**Comment by FloEdelmann** [comment]
In already well-mapped areas, most POI types will either already be well-mapped, or not at all. E.g. nearly all benches are already mapped in my vicinity, but street cabinets are almost completely missing.

**Comment by matkoniecz** [comment]
Also, in some cases I actually map the same thing in row. For example if I map needleleaved tree it is actually fairly likely that next one will be the same object type.

**Comment by westnordost** [comment]
Yeah, but often it is not. So, the last picked values store is not the suitable mechanism for that.

One mechanism that comes to mind would be the (TBD) ability to manually pin certain presets to the top. But I am not sure how an obvious and intuitive UI for that would need to look like. (I.e. that both makes it clear to the user that this is possible but also does not clutter the view with a lot of pin-buttons)

**Comment by westnordost** [comment]
Maybe show an outlined clickable pin icon to the end of the feature 

![imagen](https://github.com/streetcomplete/StreetComplete/assets/4661658/b3675799-d2b2-4502-81cf-42c807d65a8d)

and when tapping on it, it becomes filled and will be shown at the top of the default things. This would avoid this clickable pin icon to be shown on every list item in the search.

**Comment by mnalis** [comment]
>  (Either within the box or outside the box as illustrated. Not sure what would be more clear)

That might work!

On the outside seems more reasonable to me; I do not like the practice of overloading dropdown list with extra clickable elements inside it (I expect that I will get dropdown no matter where I click inside it, and consider anything inside it just a description/decoration, not as different actionable UI elements).

**Comment by RubenKelevra** [comment]
I tend to use this feature a lot in SCEE actually. So just a couple of items is usually enough to cover my usage. I think I got 5-6 items I often add.

So a MRU list would be appreciated.

**Comment by FloEdelmann** [comment]
> So, it does not appear to me to be a useful mechanism to put whatever at the top of the list that the user just selected.

Recently there was a nice step-by-step guide in a user's OSM diary about adding fire hydrants, benches and other things with StreetComplete: https://www.openstreetmap.org/user/Tex2002ans/diary/406006

In a [comment](https://www.openstreetmap.org/user/Tex2002ans/diary/406006#comment58701), they seem to agree that making the most recently used item more easily accessible would be a good idea:

> often, you’re adding multiple of the same type of object (for example, 3 benches)

**Comment by peternewman** [comment]
> Also, in some cases I actually map the same thing in row. For example if I map needleleaved tree it is actually fairly likely that next one will be the same object type.

Or round a park there will be small clusters of benches and bins, or on a street there may be batches of bins and recycling bins.

Two possible UI considerations to resolve this in a different way:
When you go to add a Thing, it pops up the bottom bit saying Select (incidentally with a three dot menu that doesn't do anything as we've not selected an existing thing).
There is dead space next to the three dot menu, could it show three or more of the MRU symbols (a bit like building levels), tapping one would select it in the drop-down giving certainty you've actually selected the right item, but removing a few presses/giving the MRU list we're all asking for.

Secondly/alternatively, when you press Select it pops up the list of ten popular items.
Even on my tiny phone, there are space for two more items in the unused gap at the bottom of the list, I'm sure a bigger screen would have space for more (either as MRU, or just for general selection)
Equally it defaults to pre-selecting the "Type kind of place here" (incidentally that's the wrong text surely, sounds like the places overlay prompt, or are they not separate; perhaps an en-GB translation issue)
Because this is pre-selected, it pops up the keyboard which hides the bottom 4 items (on my tiny screen), only showing the top 6.
It probably makes sense to pre-select/show the keyboard for places, because hopefully we're mostly using NSI/there's generally more things to chose from/more specificity even if unbranded (e.g. fish & chips fast food rather than just fast food).

Could a similar thing be done for (at least non-branded places), although it's definitely a stronger argument that you're less likely to have two cafes next to each other, but maybe the non-MRU improvements could be made to places.
